internal/service/management_api.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	metrics "github.com/slok/go-http-metrics/metrics/prometheus"
+	"github.com/slok/go-http-metrics/middleware"
+	"github.com/slok/go-http-metrics/middleware/std"
+	"github.com/topfreegames/maestro/internal/config"
+	"github.com/topfreegames/maestro/internal/core/monitoring"
+	"github.com/topfreegames/maestro/internal/handlers"
+	api "github.com/topfreegames/maestro/protogen/api/v1"
+	"go.uber.org/zap"
+)
+
+func ProvideManagementMux(ctx context.Context, pingHandler *handlers.PingHandler) *runtime.ServeMux {
+	mux := runtime.NewServeMux()
+	api.RegisterPingHandlerServer(ctx, mux, pingHandler)
+
+	return mux
+}
+
+// RunManagementServer starts HTTP server in other goroutine, and returns a
+// shutdown function. It serves management API endpoints/handlers.
+func RunManagementServer(ctx context.Context, configs config.Config, mux *runtime.ServeMux) func() error {
```

Comment:
This function looks like it could be used to start other APIs. WDYT about making it more generic?

Reply:
Actually it depends a lot on management_api configurations... don't think it could be so reusable as we would like.. also, once /internal/service is shared by all CMDs I'm tending to move this function to the CMD.

Reply:
In that case, it makes sense to move it to the CMD.